README.md

```diff
@@ -89,3 +89,162 @@ func (s *Service) CommonName(ctx context.Context, _ *CommonNameRequest) (*Common
 	return &CommonNameResponse{CommonName: *commonName}, nil
 }
 ```
+
+
+## TODOs
+- [ ] Setting cookies in gRPC-web: https://github.com/improbable-eng/grpc-web/issues/833
+- [ ] Support PKCE with OIDC
+- [ ] Adhere to the Listen, Run, Stop interface from rpcserve
+- [ ] Logger / Error pkg compatibility
+- [ ] Make server props private
+- [ ] Transparent Re-Auth as implemented might be insecure?
+- [ ] Chainable authentication mechanisms
+
+
+## OIDC
+The server provides an implementation to support deployments where OIDC is being used 
+
+Information and design decisions about the OIDC implementation are summarized below. They are largely based on the
+following documents - have a read - it's probably worth it:
+- [Best Practices - OAuth for Single Page Apps [curity.io]](https://curity.io/resources/learn/spa-best-practices/)
+- [OAuth 2.0 for Browser-Based Apps [ietf.org]](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#section-6.2)
+- [Authorization Code Flow with Proof Key for Code Exchange [auth0.com]](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-proof-key-for-code-exchange-pkce)
+- [Proof Key for Code Exchange by OAuth Public Clients [ietf.org]](https://datatracker.ietf.org/doc/html/rfc7636)
+
+
+### BFF (Backend for Frontend)-style proxy for SPAs (Single Page Apps)
+The browser is a hostile place to execute code, and implementing security is a difficult area of SPA development.
+We therefore implement an adoption of the BFF (Backend for Frontend) pattern for SPAs according to the
+[IETF Draft "OAuth 2.0 for Browser-Based Apps" Section 6.2](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#section-6.2).
+Access Tokens and Refresh Tokens are kept secured on the server while a traditional cookie based session is initialized
+with the browser. The Cookie can be a secure http-only implementation, further reducing the attack surface. 
```

Comment:
Shouldn't the Cookie CONTAIN the access token? It sounds like we are keeping the access token server side, which I don't think is according to the standards.

Reply:
According to auth0 also the refresh token would be returned to the SPA. I wonder if we can safe that in a cookie that is bound to another domain so it is only sent on an actual refresh call. It shouldn't be sent with every call as that would put the refresh token at the same risk the access token has, just with a lot longer life-span.
> Your Auth0 Authorization Server responds with an ID token and access token (and optionally, a refresh token).

https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-proof-key-for-code-exchange-pkce